### DISCUSSION TITLE
Selected Site Twig Variable within the Control Panel

### DISCUSSION THREAD

**Discussion by richhayler** [craftcms/cms#11158] [Plugin ideas] [answered]
As of Craft 4, the selected site will be remembered across CP pages via a site query string param. You can fetch the site that param is referencing via craft\helpers\Cp::requestedSite().

`e.g. /entries/pages?site=mysite`

It would be great to make this selected site available as a twig variable within the control panel so for multi-site setups it could be directly referenced or used in queries relating to the selected site as opposed to the current site from where you were accessing the control panel, e.g.

```
{# set selectedSite variable and fallback default value to currentSite #}
{% set selectedSite = selectedSite|default(currentSite) %}

{# create an array for the fields within a globalset for the selected site #}
{% set selectedSiteGlobals = craft.globalSets().handle('globalSetHandle').site(selectedSite).all() %}

{% set selectedSiteGlobal = selectedSiteGlobals[0] %}

{# set a new variable for a specific field in the selectedSite global set#}
{% set color = selectedSiteGlobal.colorField %}

{# use the value of that variable e.g. to change the colour of the nav so it matches the colour set for the selected site #}
<style>
#nav {background-color: {{ color }}; }
</style>
```
Another example could be getting the selected site name and appending that within the control panel UI, e.g.
`:after {content: “{{ selectedSite }}”;}`
or applying other visual cues to indicate to the site author which site is currently selected.

**Comment by brandonkelly** [comment, marked as answer]
This seems identical to your previous request (#11082). Am I missing something?

**Reply by richhayler** [reply]
Yes it is identical. I thought it was more appropriate to post my request here as a plug-in/extension request rather than change to the core product based on your original feedback. Apologies for the duplication.

I wasn’t sure how to go about achieving this myself based on your previous advice. So I went  through the craft docs, nystudio107 blog and also signed up at CraftQuest to try to figure it out. I created a custom module and twig extension and am pretty close to a solution. Can close this item and then once I have something working I can post my solution as a comment attached to the original ticket if someone else could benefit from the extended functionality.
